fibheap/fibheap_test.go

```diff
@@ -0,0 +1,427 @@
+		assert.NoError(t, err)
+		assert.Equal(t, NumberSequence2Deleted3ElemSorted[i], min.Priority)
+	}
+}
+
+func TestFibHeap_Delete_EmptyHeap(t *testing.T) {
+	heap := NewFloatFibHeap()
+
+	elem := heap.Enqueue(15)
+	heap.DequeueMin()
+
+	// Heap should be empty at this point
+	err := heap.Delete(elem)
+	assert.EqualError(t, err, "Cannot delete element from an empty heap")
+}
+
+func TestFibHeap_Delete_NilNode(t *testing.T) {
+	heap := NewFloatFibHeap()
+	heap.Enqueue(1)
+	err := heap.Delete(nil)
+	assert.EqualError(t, err, "Cannot delete node: given node is nil")
+}
+
+func TestMerge(t *testing.T) {
+	heap1 := NewFloatFibHeap()
+	for i := 0; i < len(NumberSequence3); i++ {
+		heap1.Enqueue(NumberSequence3[i])
+	}
+
+	heap2 := NewFloatFibHeap()
+	for i := 0; i < len(NumberSequence4); i++ {
+		heap1.Enqueue(NumberSequence4[i])
+	}
+
+	heap, err := heap1.Merge(&heap2)
+	assert.NoError(t, err)
+
+	var min *Entry
+	for i := 0; i < len(NumberSequenceMerged3And4Sorted); i++ {
+		min, err = heap.DequeueMin()
+		assert.NoError(t, err)
+		assert.Equal(t, NumberSequenceMerged3And4Sorted[i], min.Priority)
+	}
+}
+
+func TestFibHeap_Merge_NilHeap(t *testing.T) {
+	var heap FloatingFibonacciHeap
+	heap = NewFloatFibHeap()
+	newHeap, err := heap.Merge(nil)
+	assert.EqualError(t, err, "One of the heaps to merge is nil. Cannot merge")
+	assert.Equal(t, newHeap, FloatingFibonacciHeap{})
+}
+
+// ***************
+// BENCHMARK TESTS
+// ***************
+
+/*
+Since the e.g. Enqeue operation is constant time,
+when go benchmark increases N, the prep time
+will increase linearly, but the actual operation
+we want to measure will always take the same,
+constant amount of time.
+This means that on some machines, Go Bench
+could try to exponentially increase N in order
+to decrease noise in the measurement, but it will
+get more and more noise. This can cause a system
+to run out of RAM. So be careful if you have a fast PC.
+I have removed the b.ResetTimer on constant-time
+functions to avoid this negative-feedback loop.
+*/
+
+// Runs in O(1) time
+func BenchmarkFibHeap_Enqueue(b *testing.B) {
+
+	heap := NewFloatFibHeap()
+
+	slice := make([]float64, 0, b.N)
+	for i := 0; i < b.N; i++ {
+		slice = append(slice, 2*1E10*(rand.Float64()-0.5))
+	}
+
+	for i := 0; i < b.N; i++ {
+		heap.Enqueue(slice[i])
+	}
+}
+
+// Runs in O(log(N)) time
+func BenchmarkFibHeap_DequeueMin(b *testing.B) {
+
+	heap := NewFloatFibHeap()
+
+	slice := make([]float64, 0, b.N)
+	for i := 0; i < b.N; i++ {
+		slice = append(slice, 2*1E10*(rand.Float64()-0.5))
+		heap.Enqueue(slice[i])
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		heap.DequeueMin()
+	}
+}
+
+// Runs in O(1) amortized time
+func BenchmarkFibHeap_DecreaseKey(b *testing.B) {
+	heap := NewFloatFibHeap()
+
+	sliceFlt := make([]float64, 0, b.N)
+	sliceE := make([]*Entry, 0, b.N)
+	for i := 0; i < b.N; i++ {
+		sliceFlt = append(sliceFlt, 2*1E10*(float64(i)-0.5))
+		sliceE = append(sliceE, heap.Enqueue(sliceFlt[i]))
+	}
+
```

Comment:
If you add b.ResetTimer you can narrow down the timing to the one operation you care about.